### PR TITLE
Add default subscribe status config variable

### PR DESCRIPTION
**Pull request by dodecasphere**

### CODE REVIEW

File: src/Newsletter.php [hunk 1]
```diff
@@ -211,4 +212,16 @@ protected function getSubscriberHash($email)
     {
         return $this->mailChimp->subscriberHash($email);
     }
+
+    protected function getSubscribeStatus()
+    {
+        $subscribeStatus = config('laravel-newsletter.defaultSubscribeStatus');
+
+        if ($subscribeStatus == "subscribed" || $subscribeStatus == "pending") {
```
Comment (freekmurze) [inline src/Newsletter.php]: I don't like else branches and 'OR's. Replace this by
```php 
if (! in_array($defaultSubscribeStatus, ['subscribed', 'pending'])) {
```

File: src/Newsletter.php [hunk 2]
```diff
@@ -211,4 +212,16 @@ protected function getSubscriberHash($email)
     {
         return $this->mailChimp->subscriberHash($email);
     }
+
+    protected function getSubscribeStatus()
```
Comment (freekmurze) [inline src/Newsletter.php]: rename this to `getDefaultSubscribeStatus`

File: tests/MailChimp/NewsletterTest.php [hunk 3]
```diff
@@ -264,4 +264,25 @@ public function is_can_create_a_campaign()
 
         $this->newsletter->createCampaign($fromName, $replyTo, $subject, $html, $listName, $options, $contentOptions);
     }
+
+    /** @test */
+    public function it_will_throw_an_exception_when_using_a_subscribe_status_that_is_invalid()
```
Comment (freekmurze) [inline tests/MailChimp/NewsletterTest.php]: That's a nice test. Could you add another one to make sure that when `status` gets passed in with the `$options` that I'll override the default?

**Comment by freekmurze** [comment]
Thanks for your contribution.

I've added some remarks in the code.  Could you also make sure all tests pass in Travis?

File: tests/MailChimp/NewsletterTest.php (hunk 3 above)
Comment (dodecasphere) [inline tests/MailChimp/NewsletterTest.php]: I believe that's already addressed in the test `it_can_override_the_defaults_when_subscribing_someone` on line 111, yes?
https://github.com/dodecasphere/laravel-newsletter/blob/default-status/tests/MailChimp/NewsletterTest.php#L111

**Comment by freekmurze** [comment]
I hope my remarks didn't scare you off. Your changes were good, only needed dotting some i's and crossing some t's...

**Comment by dodecasphere** [comment]
Oh, not at all. I just realized folks could use the `options` parameter in the `subscribe` method to achieve what I was initially missing (subscribe my users as pending instead of subscribed) and don't need to put the subscribe status in the config array.

Might pick it up again if I have the time to push forward on this in the future.

I will, however, likely submit a PR updating the README adding documentation around that final parameter under **Subscribing and unsubscribing**.

Thanks much!

**Pull request closed without merging**